torch/csrc/autograd/python_variable.cpp

```diff
@@ -390,9 +390,6 @@ int THPVariable_set_requires_grad(THPVariable *self, PyObject *obj)
     return -1;
   }
   var.requires_grad() = (obj == Py_True);
-  if (auto grad_accumulator = var.get()->grad_accumulator.lock()) {
-    grad_accumulator->is_executable = var.requires_grad();
-  }
```

Comment:
I'd be careful with that, because it changes the semantics. Previously, if you changed `requires_grad` on a leaf, its grad would get frozen, which is not the case now.